Sources/Rendering/Misc/SynchronizableRenderWindow/vtkObjectManager.js

```diff
@@ -275,6 +274,7 @@ function genericUpdater(instance, state, context) {
       .finally(context.end); // -> end(arraysToBind)
   }
   context.end(); // -> end(generic-updater)
+  return Promise.resolve(true);
```

Comment:
didn't you wanted to wait for `bindArrays()` line 265 to be called before resolving the promise ?

Reply:
I would think so but when I suggested it, nothing changed so I wasn't sure. At least if the API is consistent that could be fixed afterward.

Reply:
I'm confused by this as well. If `state.arrays` is truthy, then the immediate line of execution becomes the following:
1. L262: `context.start()`
2. L263: Create a promise that waits for all other promises
3. L276: `context.end()` (this runs before `bindArrays`)

My question: does the context need to be started for `bindArrays` to work? Because in this scenario, the context is closed well before the `.then` condition on L264 is executed.

One solution to this is to simply say `return Promise.all(promises)` on L263 and do a `return true` after L265.

Reply:
Assuming I'm thinking of the right thing:
start => +1 
end => -1
The in-between work will also call internally some start/end. So the context is not really closed.
 
basically the code should be like:
```
ctx.start()
for array in arrays:
   ctx.start()
   [...]
   ctx.end()
ctx.end()
```